valhalla/baldr/double_bucket_queue.h

```diff
@@ -179,10 +179,14 @@ class DoubleBucketQueue final {
    * @return Returns the bucket that the cost lies within.
    */
   bucket_t& get_bucket(const float cost) {
-    return (cost < currentcost_)
-               ? *currentbucket_
-               : (cost < maxcost_) ? buckets_[static_cast<uint32_t>((cost - mincost_) * inv_)]
-                                   : overflowbucket_;
+    return (cost < currentcost_) ? *currentbucket_
+                                 : (cost < maxcost_) ? get_bucket_inner(cost) : overflowbucket_;
+  }
+
+  inline bucket_t& get_bucket_inner(float cost) {
+    return buckets_[static_cast<uint32_t>((cost - mincost_) * inv_
+                                          // Ensure we fall on the correct side of the integer point
+                                          + 0.5)];
```

Comment:
so previously we took the floor now we round, basically this just shifts stuff by .5 to the right. does the size of `buckets_` allow for the highest cost to not go out of bounds when it gets rounded up? i know we saw some segfaults here accessing buckets that dont exist. im guessing that `cost` was less than `mincost_`? can we add a unit test that uses double bucket queue directly in such a way that it would cause the out of bounds access?

Reply:
a check to see if cost < mincost_ makes sense here (just add the label to buckets_[0]) , though not sure why that might happen. Perhaps logging some info in that (hopefully rare) case could help understand how this happens and could lead to a unit test.